src/features/antora/antoraSupport.ts

```diff
@@ -83,18 +86,124 @@ export async function getAntoraConfig (textDocument: vscode.TextDocument): Promi
   return undefined
 }
 
-export async function getAttributes (textDocument: vscode.TextDocument): Promise<{ [key: string]: string }> {
-  const antoraConfigUri = await getAntoraConfig(textDocument)
+export async function antoraConfigExists (textDocumentUri: Uri): Promise<boolean> {
+  return await getAntoraConfig(textDocumentUri) !== undefined
+}
+
+class AntoraDisabledError extends Error {
+}
+
+export async function parseAntoraConfig (textDocumentUri: Uri): Promise<{ [key: string]: any }> {
+  const antoraConfigUri = await getAntoraConfig(textDocumentUri)
   const antoraConfigPath = antoraConfigUri.fsPath
   try {
-    const doc = yaml.load(fs.readFileSync(antoraConfigPath, 'utf8'))
-    return doc.asciidoc.attributes
+    return yaml.load(fs.readFileSync(antoraConfigPath, 'utf8'))
   } catch (err) {
     console.log(`Unable to parse ${antoraConfigPath}, cause:` + err.toString())
     return {}
   }
 }
 
-export async function getValidConfig (textDocument: vscode.TextDocument): Promise<boolean> {
-  return await getAntoraConfig(textDocument) !== undefined
+export async function getAttributes (textDocumentUri: Uri): Promise<{ [key: string]: string }> {
+  const doc = await parseAntoraConfig(textDocumentUri)
+  if (doc !== {}) {
+    return doc.asciidoc.attributes
+  } else {
+    return {}
+  }
+}
+
+export async function getContentCatalog (textDocumentUri: Uri, extensionContext: ExtensionContext): Promise<ContentCatalog | undefined> {
+  try {
+    const activeAntoraConfig = await getActiveAntoraConfig(textDocumentUri, extensionContext)
+    if (activeAntoraConfig === undefined) {
+      return undefined
+    }
+    const contentSourceRootPath = path.dirname(activeAntoraConfig.fsPath)
+    const contentSourceRepositoryRootPath = workspace.getWorkspaceFolder(activeAntoraConfig).uri.fsPath
+    // https://docs.antora.org/antora/latest/playbook/content-source-start-path/#start-path-key
+    const startPath = path.relative(contentSourceRepositoryRootPath, contentSourceRootPath)
+    const playbook = {
+      content: {
+        sources: [{
+          url: contentSourceRepositoryRootPath,
+          branches: 'HEAD',
+          startPath,
+        }],
+      },
+      runtime: {},
+      site: {},
+    }
+    try {
+      const contentAggregate = await aggregateContent(playbook)
+      return classifyContent(playbook, contentAggregate)
```

Comment:
(performance): we should aggregate and classify content only once per playbook. Use event listeners (file added, renamed, removed) to update the contentCatalog.